test-js/test-rectangle.js

```diff
@@ -0,0 +1,404 @@
+const expect = chai.expect;
+
+import {
+  pointInRect,
+  centeredRect,
+  normalizeRect,
+  sectRect,
+  unionRect,
+  offsetRect,
+  scaleRect,
+  insetRect,
+  equalRect,
+  rectCenter,
+  rectSize,
+  rectFromArray,
+  rectToArray,
+  isEmptyRect,
+  rectFromPoints,
+} from "../src/fontra/client/core/rectangle.js";
+import { parametrize } from "./test-support.js";
+
+describe("pointInRect", () => {
+  const testData = [
+    [
+      { x: 0, y: 0 },
+      { xMin: -Infinity, yMin: -Infinity, xMax: Infinity, yMax: Infinity },
+      true,
+      "Rectangle should has not been bound to a limit",
+    ],
+    [
+      { x: 40, y: 40 },
+      undefined,
+      false,
+      "Should return false if rectangle is a falsy value",
+    ],
+    [{ x: 40, y: 40 }, { xMin: 0, yMin: 0, xMax: 200, yMax: 200 }, true, "Should work"],
+    [
+      { x: 40.0, y: 40.0 },
+      { xMin: 0, yMin: 0, xMax: 200, yMax: 200 },
+      true,
+      "Should work with floats",
+    ],
+  ];
+  parametrize(
+    "is point in rectangle",
+    testData,
+    ([point, rectangle, acceptance, testDescription]) => {
+      const result = pointInRect(point.x, point.y, rectangle);
+      expect(result).equals(acceptance, testDescription);
+    }
+  );
+});
+
+describe("centeredRect", () => {
+  const testData = [
+    [
+      { x: 100, y: 100 },
+      { width: 50, height: 50 },
+      {
+        xMin: 75,
+        xMax: 125,
+        yMin: 75,
+        yMax: 125,
+      },
+      "Should create a rectangle relatively to a point",
+    ],
+    [
+      { x: 100, y: 100 },
+      { width: 50 },
+      {
+        xMin: 75,
+        xMax: 125,
+        yMin: 75,
+        yMax: 125,
+      },
+      "Should create a square if an height is not given",
+    ],
+  ];
+  parametrize(
+    "Creates a centered rectagle",
+    testData,
+    ([point, sizes, acceptance, testDescription]) => {
+      const result = centeredRect(point.x, point.y, sizes.width, sizes.height);
+      expect(result).deep.equals(acceptance, testDescription);
+    }
+  );
+});
+
+describe("normalizeRect", () => {
+  const testData = [
+    [
+      { xMin: 100, yMin: 100, xMax: 0, yMax: 0 },
+      { xMin: 0, yMin: 0, xMax: 100, yMax: 100 },
+    ],
+  ];
+  parametrize(
+    "Normalizes a rectangle",
+    testData,
+    ([rectangle, acceptance, testDescription]) => {
+      const result = normalizeRect(rectangle);
+      expect(result).deep.equals(acceptance, testDescription);
+    }
+  );
+});
+
+describe("sectRect", () => {
+  const testData = [
+    [
+      { xMin: 50, yMin: 50, xMax: 100, yMax: 100 },
+      { xMin: 75, yMin: 75, xMax: 80, yMax: 80 },
+      { xMin: 75, yMin: 75, xMax: 80, yMax: 80 },
+    ],
+    [
+      { xMin: 50, yMin: 50, xMax: 60, yMax: 60 },
+      { xMin: 60, yMin: 60, xMax: 70, yMax: 70 },
+      { xMin: 60, yMin: 60, xMax: 60, yMax: 60 },
+      "Creates rectangle with sizes of 0 if they intersect at the edge", // should be clarified with @justvanrossum
```

Comment:
@justvanrossum Should this be fixed in code?

Reply:
It's this line in the code, it should use `>=` instead of `>`:
https://github.com/googlefonts/fontra/blob/952497b65479f6adec8b1d24a93ae59d8bc5b405/src/fontra/client/core/rectangle.js#L47

It comes from fonttools, which indeed uses `>=`. I probably made a mistake when porting this code.
https://github.com/fonttools/fonttools/blob/fb56e7b7c9715895b81708904c840875008adb9c/Lib/fontTools/misc/arrayTools.py#L209

So yes, please fix.